hoomd/operations.py

```diff
@@ -1,47 +1,112 @@
-    def __init__(self, simulation=None):
-        self._simulation = simulation
+class Operations(Collection):
+    """A mutable collection of operations which act on a `hoomd.Simulation`.
+
+    The `Operations` class contains all the operations acting on a
+    simulation. These operations are classes that perform various actions on a
+    `hoomd.Simulation`. Operations can be added and removed at any point from an
+    `hoomd.Operations` instance. The class provides the interface define by
+    `collections.abc.Collection`. Other methods for manipulating instances
+    attempt to mimic Python objects where possible, but the class is not
+    simply a mutable list or set. Since there are multiple types of operations
+    in HOOMD-blue,  `Operations` objects manage multiple independent
+    sequences described below.
+
+    The types of operations which can be added to an `Operations` object are
+    integrators, updaters, analyzers, tuners, and computes. An `Operations` can
+    only ever hold one integrator at a time. On the other hand, an `Operations`
+    object can hold any number of updaters, analyzers, tuners, or computes. To
+    see examples of these types of operations see `hoomd.hpmc.integrate`
+    or `hoomd.md.integrate` (integrators), `hoomd.update` (updaters) ,
+    `hoomd.tune` (tuners), `hoomd.dump` (analyzers), and `hoomd.md.thermo`
+    (computes).
+
+    A given instance of an operation class can only be added to a single
+    `Operations` container. Likewise, a single instance cannot be added to the
+    same `Operations` container more than once.
+
+    Note:
+        An `Operations` object is created by default when a new simulation is
+        created.
+    """
+
+    def __init__(self):
         self._compute = list()
         self._scheduled = False
-        self._updaters = SyncedList(OnlyType(_Updater),
+        self._updaters = SyncedList(OnlyType(Updater),
                                     _triggered_op_conversion)
-        self._analyzers = SyncedList(OnlyType(_Analyzer),
+        self._analyzers = SyncedList(OnlyType(Analyzer),
                                      _triggered_op_conversion)
-        self._tuners = SyncedList(OnlyType(_Tuner), lambda x: x._cpp_obj)
+        self._tuners = SyncedList(OnlyType(Tuner), lambda x: x._cpp_obj)
         self._integrator = None
 
         self._tuners.append(ParticleSorter())
 
-    def add(self, op):
+    def add(self, operation):
+        """Add operation to this container.
+
+        Adds the provide operation to the appropriate attribute of the
+        `Operations` instance.
+
+        Args:
+            operation (``operation``): A HOOMD-blue updater, analyzers, compute,
+                tuner, or integrator to add to the collection.
+
+        Raises:
+            RuntimeError: raised when an operation belonging to to this or
+                another `Operations` instance is passed.
+            TypeError: raised when the passed operation is not of a valid type.
```

Comment:
I'm adopting the "Raises" phrasing structure from [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) in this suggestion. I prefer its conciseness. Also I think the `RuntimeError` should be a `ValueError` instead. It's an invalid value. Examples of `RuntimeError`s I can come up with seem more obscure and are usually not related to adding/setting a value.
```suggestion
            ValueError: If ``operation`` already belongs to this or another `Operations` instance.
            TypeError: If ``operation`` is not of a valid type.
```